app/src/main/java/com/realityexpander/tasky/di/AppModule.kt

```diff
@@ -42,4 +107,100 @@ object AppModule {
     @Singleton
     fun provideValidateUsername(): ValidateUsername = ValidateUsername()
 
-}
\ No newline at end of file
+    @Provides
+    @Singleton
+    fun provideAuthRepository(): IAuthRepository {
+        if (USE_FAKE_REPOSITORY) {
+            return provideAuthRepositoryFake(
+                authApi = provideAuthApiFakeImpl(),
+                authDao = AuthDaoFakeImpl(),
+                validateUsername = provideValidateUsername(),
+                validatePassword = provideValidatePassword(),
+                validateEmail = provideValidateEmailRegex(),
+            )
+        } else {
+            return provideAuthRepositoryProd(
+                authApi = provideAuthApiProd(
+                    provideTaskyApi()
+                ),
+                authDao = AuthDaoFakeImpl(),  // why cant we use the implementation from @Binds here?
+                validateUsername = provideValidateUsername(),
+                validatePassword = provideValidatePassword(),
+                validateEmail = provideValidateEmailRegex(),
+            )
+        }
+    }
```

Comment:
Idk, if I add `@AuthDaoFakeUsingBinds fakeDao: IAuthDao` as a parameter here, your app launches. Looks fine to me.

Reply:
My question is specifically: For `@Bind`ing an interface, is there a way to pass a parameter using the `@Binds` annotation, or is the only way to pass an argument to an interface is to have hilt instantiate it with a `@Provides` and pass the parameter there?

Reply:
If your `AuthApiImpl` has an inject constructor that takes a `TaskyApi`, Hilt should actually automatically create that with your provided API interface. If there are multiple, it will complain (not sure if you can use `@Named` here in Binds). 
Either way, you also don't do anything wrong with using Provides for everything. Generates a bit more code, but has no performance drawbacks.